pkg/unions/kubernetes_ingress.go

```diff
@@ -0,0 +1,35 @@
+package unions
+
+import (
+	"github.com/devfile/library/pkg/devfile/generator"
+	"github.com/openshift/odo/pkg/odogenerator"
+	"k8s.io/api/extensions/v1beta1"
+	"k8s.io/api/networking/v1"
+)
+
+type KubernetesIngress struct {
+	NetworkingV1Ingress     *v1.Ingress
+	ExtensionV1Beta1Ingress *v1beta1.Ingress
+	isGenerated             bool
+}
+
+func NewEmptyKubernetesIngress() *KubernetesIngress {
+	return &KubernetesIngress{
+		NetworkingV1Ingress:     nil,
+		ExtensionV1Beta1Ingress: nil,
+		isGenerated:             false,
+	}
+}
+
+func NewKubernetesIngressFromParams(ingressParams generator.IngressParams) *KubernetesIngress {
+	ki := &KubernetesIngress{
+		NetworkingV1Ingress:     odogenerator.GetNetworkingV1Ingress(ingressParams),
+		ExtensionV1Beta1Ingress: generator.GetIngress(ingressParams),
+		isGenerated:             true,
+	}
+	return ki
+}
+
+func (ki *KubernetesIngress) IsGenerated() bool {
```

Comment:
Do we need a Getter, or can we expose the member IsGenerated directly?

Reply:
I think having getter is better here as I do not want the member to be set accidentally. Only the `NewKubernetesIngressFromParams` should set i`sGenerated` to `true`